workflows/eager.nf

```diff
@@ -215,6 +217,33 @@ workflow EAGER {
         ch_dedupped_flagstat = Channel.empty()
     }
 
+    //
+    // Subworkflow: Metagenomics screening
+    //
+
+    if( params.run_metagenomicscreening ) {
+        ch_bamfiltered_for_metagenomics.map{ meta, fastq ->
+            [meta+['single_end':true], fastq]
+        }
+        .set{ch_bamfiltered_for_metagenomics}
+
+        // Is a complexity filter wanted?
+        if ( params.run_metagenomics_complexity ) {
+            // pick the complexity tool
+            if (params.metagenomics_complexity_tool == 'bbduk') {
+                BBMAP_BBDUK( ch_bamfiltered_for_metagenomics, [] )
+                ch_versions = ch_versions.mix( BBMAP_BBDUK.out.versions.first() )
+                ch_reads_for_metagenomics = BBMAP_BBDUK.out.reads
+            }
+            else if ( params.metagenomics_complexity_tool == 'prinseq' ) {
+                PRINSEQPLUSPLUS ( ch_bamfiltered_for_metagenomics )
+                ch_versions = ch_versions.mix( PRINSEQPLUSPLUS.out.versions )
+                ch_reads_for_metagenomics = PRINSEQPLUSPLUS.out.good_reads
+            }
+        } else {
+            ch_reads_for_metagenomics = ch_bamfiltered_for_metagenomics
+        }
```

Comment:
Should be moved into an entirely separate subworkflow, within which the actaul profilers will also be included

Reply:
We will split the profiling and the filtering into two separate subworkflows, so that in the main eager.nf workflow it is easier to comprehend. Complexity filter yes? --> run filter subworkflow, then run profiling workflow

Reply:
Or metagenomics subworkflow, inside you have two or three subworkflows -> complexity filtering / profiling / post-profiling (for hops etc)

Reply:
Sounds also good :)

Reply:
is kraken parse considered post-profiling?

Reply:
I think so